cpp/oneapi/dal/io/csv/detail/read_graph_kernel_impl.hpp

```diff
@@ -39,28 +39,38 @@ inline void read_edge_list(const std::string &name, edge_list<std::int32_t> &eli
     if (!file.is_open()) {
         throw invalid_argument(dal::detail::error_messages::file_not_found());
     }
-
-    elist.reserve(1024);
-    char source_vertex[32], destination_vertex[32];
-    while (file >> source_vertex >> destination_vertex) {
-        auto edge = std::make_pair(daal_string_to_int(&source_vertex[0], 0),
-                                   daal_string_to_int(&destination_vertex[0], 0));
-
-        if (edge.first == 0 && source_vertex[0] != '0') {
-            throw invalid_argument(
-                dal::detail::error_messages::non_numeric_character_in_edge_list());
+    elist.reserve(1024); 
+    std::string line;
+    char * source_endptr;
+    char * dest_endptr;
+    const char *edgeline;
+    std::int32_t source_vertex;
+    std::int32_t destination_vertex;
+    while (getline(file, line)) {      
+        if (line.empty()){
+            break;
```

Comment:
Thank you for pointing that out! This condition was used to avoid incorrect processing of the last (empty) line of the file. I think break statement should be replaced with continue to satisfy both the case you described and the case of the last line